Updated to support Apple's new Binary Interface in order to send a priority item.

gwynantj opened the pull request:
## Description
- The current version was using an old format to send a push notification but in order to send silent notifications we need to set the priority item to 5.
- This PR is based off an existing PR here: https://github.com/tompesman/push-apns/pull/8
- The documentation for the binary format is covered here: https://developer.apple.com/library/ios/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/Appendixes/BinaryProviderAPI.html#//apple_ref/doc/uid/TP40008194-CH106-SW5
- Added tests to verify that the binary format is correct

gwynantj committed:
Fixed the binary notification format to use the new one.

biinari commented:
For a PR to upstream I would stick to their code style for quotes etc. Also worth squashing the commits down to remove the unnecessary history of 'fixed this'.

biinari commented on Gemfile:
```diff
@@ -13,4 +13,4 @@ gemspec
 
-gem 'push-core', :path => "~/code/push/push-core"
\ No newline at end of file
+gem 'push-core', path: '~/code/push/push-core'
```
Would leave this alone as it is not relevant to this PR.

biinari commented:
If you fork from the branch in the original PR then the history would correctly show authorship.

gwynantj commented:
I don't have permission to fork repos to the Fatsoma organisation, do you want me to do that? But if I squash the commits after that will it still show correct authorship?

gwynantj commented:
@biinari I've added tests. Not sure about forking off the original PR......since the user that opened the PR hasn't responded to any of the original comments.

biinari commented:
ok @gwynantj no worries about forking off the original PR then.

biinari commented on spec/push/message_apns_spec.rb:
```diff
@@ -0,0 +1,135 @@
+  subject { Push::MessageApns.create(notification_options) }
+
+  describe "#badge" do
+    subject { super().badge }
```
Not a fan of super subjects. I would have a let instead of the subject above as `let(:instance) { Push::MessageApns.create(notification_options) }` and refer to `instance` in subjects below

biinari commented on push-apns.gemspec:
```diff
@@ -20,4 +20,7 @@ Gem::Specification.new do |s|
 
   s.add_dependency "push-core", "~> 1.0.0"
   s.add_development_dependency "sqlite3"
+  s.add_development_dependency "rspec"
```
Would specify RSpec 3.x:

``` ruby
s.add_development_dependency "rspec", "~>3.4"
```

biinari commented on spec/push/message_apns_spec.rb:
```diff
@@ -0,0 +1,135 @@
+   end
+
+   it "should include five items" do
+     _1, _2, items_stream = subject.to_message.unpack("cNa*")
```
Common pattern here could be extracted to a method or even into a custom matcher

biinari commented on spec/spec_helper.rb:
```diff
@@ -0,0 +1,25 @@
+require "active_record"
+require "push-core"
+require "multi_json"
+require "./lib/push-apns"
```
would avoid relative requires like this. Instead use:

``` ruby
require "bundler/setup"
require "push-apns"
```

biinari commented on spec/spec_helper.rb:
```diff
@@ -0,0 +1,25 @@
+require "./lib/push-apns"
+
+ActiveRecord::Base.establish_connection(:adapter => "sqlite3", :database => ":memory:")
+ActiveRecord::Schema.define(:version => 1) do
```
This would need manually updating if any schema changes occur. Any way we could load this up from the generators (perhaps using a dummy app in `spec/dummy`)

biinari commented on spec/push/message_apns_spec.rb:
```diff
@@ -0,0 +1,135 @@
+describe Push::MessageApns do
+  let(:notification_options) {
+    {
+      app: 'com.test.app',
```
I think it would be worth generating some random values for these and referring to them separately. That could either be in separate `let` statements for each element, or just within this hash and refer to them below as `notification_options[:app]` etc.

gwynantj commented:
@biinari I've added a dummy rails app to get rid of the manual table creation.

I've also updated the tests with your suggestions except for the random values, not quite sure what the benefit would be in doing this (all this is still pretty new to me).

I can squash this down to 1 commit once you're happy with the changes.
